test/widgets/topic_list_test.dart

```diff
@@ -146,25 +146,30 @@ void main() {
     await tester.pump();
 
     // Tap "TOPICS" button navigating to the topic-list page…
-    connection.prepare(json: GetStreamTopicsResult(
-      topics: [eg.getStreamTopicsEntry(name: 'topic A')]).toJson());
+    connection.prepare(json: GetChannelTopicsResult(
+      topics: [eg.getChannelTopicsEntry(name: 'topic A')]).toJson());
     await tester.tap(find.byIcon(ZulipIcons.topics));
     await tester.pump();
     await tester.pump(Duration.zero);
+    check(connection.takeRequests())
+    ..length.equals(2) // one for the messages request, another for the topics
```

Comment:
This can be made a bit cleaner by turning this comment into self-checking code: put another `connection.takeRequests` call sooner, after the messages request, and just check there that there's 1 request. (Or not even check that, since the point of this test is about the requests on the topic-list page.)

That way this one covers only the request(s) made at this step of navigating to the topic-list page, and it can make a stronger check by saying there's just this one request.